src/main/java/edu/cornell/kfs/module/purap/document/service/impl/CuRequisitionServiceImpl.java

```diff
@@ -82,115 +85,144 @@ protected String checkAutomaticPurchaseOrderRules(RequisitionDocument requisitio
+                PurchaseOrderDocument.class, PurapRuleConstants.ITEMS_REQUIRE_COMMODITY_CODE_IND);
         boolean commodityCodeRequired = "Y".equals(purchaseOrderRequiresCommodityCode);
         
-        for (Iterator iter = requisition.getItems().iterator(); iter.hasNext();) {
-            RequisitionItem item = (RequisitionItem) iter.next();
+        for (Object anItem : requisition.getItems()) {
+            RequisitionItem item = (RequisitionItem) anItem;
             if (item.isItemRestrictedIndicator()) {
                 return "Requisition contains an item that is marked as restricted.";
             }
-            
+
             //We only need to check the commodity codes if this is an above the line item.
             if (item.getItemType().isLineItemIndicator()) {
-                String commodityCodesReason = "";
-                List<VendorCommodityCode> vendorCommodityCodes = commodityCodeRequired ? requisition.getVendorDetail().getVendorCommodities() : null;
-                commodityCodesReason = checkAPORulesPerItemForCommodityCodes(item, vendorCommodityCodes, commodityCodeRequired);
+                List<VendorCommodityCode> vendorCommodityCodes = commodityCodeRequired ?
+                        requisition.getVendorDetail().getVendorCommodities() : null;
+                String commodityCodesReason = checkAPORulesPerItemForCommodityCodes(item, vendorCommodityCodes,
+                        commodityCodeRequired);
                 if (StringUtils.isNotBlank(commodityCodesReason)) {
                     return commodityCodesReason;
                 }
             }
             
-            if (PurapConstants.ItemTypeCodes.ITEM_TYPE_ORDER_DISCOUNT_CODE.equals(item.getItemType().getItemTypeCode())
-                    || PurapConstants.ItemTypeCodes.ITEM_TYPE_TRADE_IN_CODE.equals(item.getItemType().getItemTypeCode())) {
-                if ((item.getItemUnitPrice() != null) && ((BigDecimal.ZERO.compareTo(item.getItemUnitPrice())) != 0)) {
+            if (PurapConstants.ItemTypeCodes.ITEM_TYPE_ORDER_DISCOUNT_CODE.equals(
+                        item.getItemType().getItemTypeCode())
+                    || PurapConstants.ItemTypeCodes.ITEM_TYPE_TRADE_IN_CODE.equals(
+                            item.getItemType().getItemTypeCode())) {
+                if (item.getItemUnitPrice() != null && BigDecimal.ZERO.compareTo(item.getItemUnitPrice()) != 0) {
                     // discount or trade-in item has unit price that is not empty or zero
-                    return "Requisition contains a " + item.getItemType().getItemTypeDescription() + " item, so it does not qualify as an APO.";
+                    return "Requisition contains a " + item.getItemType().getItemTypeDescription() +
+                            " item, so it does not qualify as an APO.";
                 }
             }
-            
-            
+//            //Base code logic check not in CU mod
```

Comment:
should this commented out code remain?

Reply:
The purpose of these changes are to start getting us back to base code.
When the next mod in the purap area needs to be removed, IMHO is will be easier to pinpoint base code versus CU code in this method if these comments are left.

Reply:
hmmm, I guess we can try and see how it goes